### PR TITLE
lint fixes from @CaraWang

### PR DESCRIPTION
**Pull request by Tristan-Wilson**
This PR cherry picks most of the lint fixes from https://github.com/OffchainLabs/go-ethereum/pull/208 from @CaraWang, with some manual conflict resolution done.

The following table shows the commits from the original PR, and whether they were included or not. 

```
| Title                                                                 | Commit  | Status                                           |
|-----------------------------------------------------------------------+---------+--------------------------------------------------|
| core/state: check amount object existence                             | 3017631 | cherry-picked                                    |
| eth/tracers: pass config to callTracer                                | e6e6f4d | cherry-picked                                    |
| cmd/evm: add L1 gas used                                              | 22f9f5a | already fixed on master                          |
| rpc: fix SA1019: jwt.StandardClaims is deprecated                     | 7298902 | jwt code changed on master, no longer applicable |
| metrics, core: fix G601 and goimports issues                          | ae568b6 | cherry-picked                                    |
| *: fix S1009, S1023, and SA4023 lint issues                           | 1eee4fe | cherry-picked                                    |
| internal/ethapi: implement SyncProgressMap and FallbackClient for bac | 6ea71f0 | already fixed on master                          |
| *: fix misspelling and unnecessary conversion                         | e994cec | cherry-picked, resolved conflicts                |
| *: fix unnecessary leading newline                                    | a75a204 | cherry-picked, resolved conflicts                |
| core: use DefaultTriesInMemory in test                                | c9da1e6 | already fixed on master                          |
```

### CODE REVIEW

File: core/state/statedb.go [hunk 1]
```diff
@@ -430,9 +430,11 @@ func (s *StateDB) SubBalance(addr common.Address, amount *big.Int) {
 func (s *StateDB) SetBalance(addr common.Address, amount *big.Int) {
 	stateObject := s.GetOrNewStateObject(addr)
 	if stateObject != nil {
-		prevBalance := stateObject.Balance()
-		s.unexpectedBalanceDelta.Add(s.unexpectedBalanceDelta, amount)
-		s.unexpectedBalanceDelta.Sub(s.unexpectedBalanceDelta, prevBalance)
+		if amount != nil {
+			prevBalance := stateObject.Balance()
+			s.unexpectedBalanceDelta.Add(s.unexpectedBalanceDelta, amount)
+			s.unexpectedBalanceDelta.Sub(s.unexpectedBalanceDelta, prevBalance)
```
Comment (PlasmaPower) [inline core/state/statedb.go:436]: I'm curious how this got hit, but I think we want to treat amount as zero if it's nil for the purposes of the `unexpectedBalanceDelta` computation. That's what `newStateObject` seems to do.

Comment (Tristan-Wilson) [reply core/state/statedb.go:436]: Changed it to treat nil amount as zero.

**Review by tsahee — COMMENTED** [review]
LGTM

**Comment by Tristan-Wilson** [comment]
@CaraWang For our Nitro codebase we require contributors to sign our Contributor License Agreement, and it covers our other repositories as well like this one. We are awaiting legal review to see if the CLA is needed in this case- it may turn out to be unneeded for this repo. In the meantime, if you would like us to be able to merge your fixes sooner, please sign the linked documents below to get yourself added to the list of approved contributors. https://na3.docusign.net/Member/PowerFormSigning.aspx?PowerFormId=b15c81cc-b5ea-42a6-9107-3992526f2898&env=na3&acct=6e152afc-6284-44af-a4c1-d8ef291db402&v=2

**Review by PlasmaPower — APPROVED** [review]
LGTM

**Pull request merged**